org.eclipse.corrosion/src/org/eclipse/corrosion/debug/RustDebugTabGroup.java

```diff
@@ -29,17 +29,36 @@
 
 	@Override
 	public void createTabs(ILaunchConfigurationDialog arg0, String arg1) {
-		setTabs(new ILaunchConfigurationTab[] { new RustDebugTab(), new CArgumentsTab(), new EnvironmentTab(),
-				new RustLocalApplicationCDebuggerTab(), new SourceLookupTab(), new CommonTab() });
+		RustDebugTab rustDebugTab = new RustDebugTab();
+		RustLocalApplicationCDebuggerTab rustLocalApplicationCDebuggerTab = new RustLocalApplicationCDebuggerTab();
+
+		rustDebugTab.addProjectInputListener(rustLocalApplicationCDebuggerTab);
+
+		setTabs(new ILaunchConfigurationTab[] { rustDebugTab, new CArgumentsTab(), new EnvironmentTab(),
+				rustLocalApplicationCDebuggerTab, new SourceLookupTab(), new CommonTab() });
 	}
 
-	protected class RustLocalApplicationCDebuggerTab extends LocalApplicationCDebuggerTab {
+	protected class RustLocalApplicationCDebuggerTab extends LocalApplicationCDebuggerTab
+			implements ProjectInputListener {
 
 		@Override
 		public void setDefaults(ILaunchConfigurationWorkingCopy config) {
 			super.setDefaults(config);
 			config.setAttribute(ICDTLaunchConfigurationConstants.ATTR_DEBUGGER_STOP_AT_MAIN, false);
 			config.setAttribute(IGDBLaunchConfigurationConstants.ATTR_DEBUG_NAME, "rust-gdb"); //$NON-NLS-1$
 		}
+
+		@Override
+		public void projectChanged(String projectName) {
+			if (false == projectName.equalsIgnoreCase(".cargo")) { //$NON-NLS-1$
+				fStopInMainSymbol.setText(projectName + "::main"); //$NON-NLS-1$
+			}
+		}
+
+	}
+
+	protected interface ProjectInputListener {
```

Comment:
Can we avoid adding a new listener class here if we have only one internal use-case for it? It seems to me like over-engineering.